lib/Common/CommonDefines.h

```diff
@@ -140,7 +140,11 @@
 #define ENABLE_JS_ETW                               // ETW support
 #else
 #define SYSINFO_IMAGE_BASE_AVAILABLE 0
+#ifndef ENABLE_VALGRIND
 #define ENABLE_CONCURRENT_GC 1
+#else
+#define ENABLE_CONCURRENT_GC 0
```

Comment:
> I think you're right that the 32GB approach might not work if under valgrind doesn't guarantee that the memory range is contiguous and starts from 0

@digitalinfinity thanks for the details there. IIRC, process stack is part of user mode space and accommodated under .text section hence it won't start from 0. I was (initially) interested implementing a smaller card table idea. However the inconsistency among the stack and process address space require changes to card table indexing approach. Besides, we will have to limit stack space too.

This PR should just work. I don't think user will need Concurrent GC for Valgrind. Only pain here is to maintain `--valgrind` flag on a host project.